backend/src/git.rs

```diff
@@ -113,15 +113,11 @@ impl Interface {
     }
 
     /// Create or overwrite the document at the provided `path` and populate it with the value of `new_doc`.
-    /// `message` will be included in the commit message, and `token` is a valid github auth token.
-    ///
-    /// # Panics
-    /// This function will panic if it's called when the repo mutex is already held by the current thread
+    /// `message` will be included in the commit message, and `branch` specifies which branch to commit to.
+    /// `token` is a valid github auth token.
     ///
     /// # Errors
-    /// This function will return an error if filesystem operations fail, or if any of the git operations fail
-    // This lint gets upset that `repo` isn't dropped early because it's a performance heavy drop, but when applied,
-    // it creates errors that note the destructor for other values failing because of it (tree)
```

Comment:
This comment was removed, but the associated function annotation for that lint is still present.